### PR TITLE
ZK-4543: Calendar should respect the constraint while Month changing

### PR DESCRIPTION
**Pull request by b952426**
ZK-4543: Calendar should respect the constraint while Month changing

### CODE REVIEW

File: zul/src/archive/web/js/zul/db/Calendar.js [hunk 1]
```diff
@@ -744,7 +750,9 @@ zul.db.Calendar = zk.$extends(zul.Widget, {
 	 * @return Date
 	 */
 	getTime: function () {
-		return this._value || zUtl.today(this.getFormat(), _getTimeZone(this));
+		var today = zUtl.today(this.getFormat(), _getTimeZone(this));
+		if(this._selectedValue == null) this._selectedValue = today;
```
Comment (b952426) [inline zul/src/archive/web/js/zul/db/Calendar.js:754]: if ()

File: zul/src/archive/web/js/zul/db/Calendar.js [hunk 2]
```diff
@@ -836,7 +845,10 @@ zul.db.Calendar = zk.$extends(zul.Widget, {
 		var newTime = _newDate(year, month, day, !nofix, tz);
 		if (!ignoreUpdate) {
 			this._value = newTime;
-			this.fire('onChange', {value: this._value, shallClose: false, shiftView: true});
+			if (!this._isSelectDisabled) {
```
Comment (b952426) [inline zul/src/archive/web/js/zul/db/Calendar.js:848]: var _isSelectDisabled? or use Renderer.disabled()

File: zul/src/archive/web/js/zul/db/Calendar.js [hunk 3]
```diff
@@ -1094,6 +1109,10 @@ zul.db.Calendar = zk.$extends(zul.Widget, {
 								if (Renderer.disabled(this, y, m + monofs, v, today)) {
 									$cell.addClass(disdClass);
 								}
+								if (selCell && selCell.hasClass(disdClass)) {
+									selCell.removeClass(seldClass);
+									selCell = '';
```
Comment (b952426) [inline zul/src/archive/web/js/zul/db/Calendar.js:1114]: selCell = null;

File: zul/src/archive/web/js/zul/db/Calendar.js [hunk 4]
```diff
@@ -1082,8 +1095,10 @@ zul.db.Calendar = zk.$extends(zul.Widget, {
 							var $cell = jq(week.cells[k]);
```
Comment (b952426) [inline zul/src/archive/web/js/zul/db/Calendar.js:1096]: var disabledCell = Renderer.disabled(this, y, m + monofs, v, today);

File: zul/src/archive/web/js/zul/db/Calendar.js [hunk 5]
```diff
@@ -712,6 +715,7 @@ zul.db.Calendar = zk.$extends(zul.Widget, {
 		this._setView('day');
 	},
 	_shiftView: function (ofs, disableAnima) {
+		var selDate = this.getTime();
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/db/Calendar.js:718]: unused var

File: zul/src/archive/web/js/zul/db/Calendar.js [hunk 6]
```diff
@@ -739,6 +743,13 @@ zul.db.Calendar = zk.$extends(zul.Widget, {
 		this._shiftView(intDelta > 0 ? -1 : 1, true);
 		evt.stop();
 	},
+	/** Returns the Date that is selected.
+	 *  <p>returns today if value is null
+	 * @return Date
+	 */
+	_getSelectedValue: function () {
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/db/Calendar.js:750]: remove this unused method

**Review by scribetw — CHANGES_REQUESTED** [review]

**Review by plijyfes — APPROVED** [review]

**Review by scribetw — APPROVED** [review]

**Comment by scribetw** [comment]
@zkadmin rebase

**Review by DevChu — APPROVED** [review]

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**